diff --git a/agent/bin/Makefile b/agent/bin/Makefile
index c2a69fd..df021fb 100644
--- a/agent/bin/Makefile
+++ b/agent/bin/Makefile
@@ -49,3 +49,6 @@ endif
 
 # exclude libraries which are not required
 LIBS := $(filter -lpgport -lpgcommon, $(LIBS))
+
+# fix upstream issues on the package for debian-based systems
+PG_LIBS += -L$(shell $(PG_CONFIG) --pkglibdir)
diff --git a/agent/lib/Makefile b/agent/lib/Makefile
index 52475de..50ea3f0 100644
--- a/agent/lib/Makefile
+++ b/agent/lib/Makefile
@@ -34,3 +34,6 @@ subdir = contrib/$(MODULE_big)
 include $(makefile_global)
 include $(top_srcdir)/contrib/contrib-global.mk
 endif
+
+# fix upstream issues on the package for debian-based systems
+PG_LIBS += -L$(shell $(PG_CONFIG) --pkglibdir)
diff --git a/reporter/Makefile b/reporter/Makefile
index fbee435..3649f7b 100644
--- a/reporter/Makefile
+++ b/reporter/Makefile
@@ -38,3 +38,6 @@ endif
 
 # exclude libraries which are not required
 LIBS := $(filter -lpgport -lpgcommon, $(LIBS))
+
+# fix upstream issues on the package for debian-based systems
+PG_LIBS += -L$(shell $(PG_CONFIG) --pkglibdir)
